src/internal/pachd/restore.go

```diff
@@ -26,9 +30,28 @@ func NewRestoreSnapshot(env RestoreSnapshotEnv, config pachconfig.PachdRestoreSn
 		printVersion(),
 		awaitDB(env.DB),
 		setupStep{
-			Name: "HelloWorld",
+			Name: "restore",
 			Fn: func(ctx context.Context) error {
-				log.Info(ctx, "Hello, you would have just restored a snapshot")
+				//tracker := track.NewPostgresTracker(env.DB)
+
+				bucket, err := obj.NewBucket(ctx, config.StorageBackend, config.StorageRoot, config.StorageURL)
+				if err != nil {
+					return errors.Wrap(err, "storage env")
+				}
+				sEnv := storage.Env{
+					Bucket: bucket,
+					DB:     env.DB,
+					Config: config.StorageConfiguration,
+				}
+				storage, err := storage.New(ctx, sEnv)
+				if err != nil {
+					return errors.Wrapf(err, "could not configure new storage")
+				}
+				s := &recovery.Snapshotter{DB: env.DB, Storage: storage.Filesets}
+				snapshotID := recovery.SnapshotID(config.SnapshotID)
+				if err := s.RestoreSnapshot(ctx, snapshotID, recovery.RestoreSnapshotOptions{}); err != nil {
+					return errors.Wrapf(err, "could not restore snapshot %s", snapshotID)
+				}
```

Comment:
I think we should call admindb.RestartPachyderm here.  Pipelines could still be running and that will get rid of them.  (starting up pachd should also do that, but probably only if the version number changes.)